pages/blog/components/Article.tsx

```diff
@@ -16,24 +16,28 @@ const ArticleWithContent = () => {
 }
 
 const Article = () => {
-  const { router } = useRouter()
+  const router = useRouter()
 
   console.log('router is ', router)
+
+  // ts-ignoring react-contentful type errors for now because their type definitions are too generic
   const { data, error, loading } = useContentful({
     contentType: 'blogPost',
   })
 
   if (loading) {
-    return 'Loading...'
+    return <>Loading...</>
   } else if (error) {
     return (
       <pre>
+        {/* @ts-ignore */}
         <code>{error}</code>
       </pre>
     )
   } else if (data) {
     return (
       <>
+        {/* @ts-ignore */}
```

Comment:
I think `ts-ignore` should be avoided at all costs, I'd rather have `any`.

Reply:
What is the TS Error here? If the library doesn't have a good definition available, we can provide "good enough" definitions through casting.

Reply:
> What is the TS Error here?

"TS2339: Property 'items' does not exist on type 'object'". Basically both "data" and "error" are typed as object in react-contentful types... 😞 

```tsx
export interface HookResponse {
  data?: object;
  error?: object;
  fetched: boolean;
  loading: boolean;
}
```

> If the library doesn't have a good definition available, we can provide "good enough" definitions through casting.

I wasn't sure what would be your preferred approach and thought it would be better to discuss this first (hence, the `@ts-ignore` "for now"). In this case the library exposes the return type as an interface, so we can tweak it to our needs via module augmentation. Does this sound good?

Reply:
@pastelsky, please take another look once you get a chance. I went ahead and augmented `react-contentful` types, as well as removed all `@ts-ignore`s